scale-tests/kafka_cassandra_streaming_test.py

```diff
@@ -175,58 +202,70 @@ def append_submission(output_file: str, dispatcher_app_name: str, submission_id:
 
     with open(args["<infrastructure_file>"]) as f:
         infrastructure = json.loads(f.read())
-        # We might make this script handle multiple Zookeeper/Kafka/Cassandra
-        # clusters in the future. For now it only handles a single cluster of
-        # each service.
-        zookeeper = infrastructure['zookeeper'][0]
-        kafka = infrastructure['kafka'][0]
+        zookeepers = infrastructure['zookeeper']
+        kafkas = infrastructure['kafka']
         cassandra = infrastructure['cassandra'][0]
 
     jar                           = args["--jar"] if args["--jar"] else DEFAULT_JAR
     submissions_output_file       = args["<submissions_output_file>"]
-    kafka_package_name            = kafka['package_name']
-    kafka_service_name            = kafka['service']['name']
-    kafka_num_brokers             = kafka['brokers']['count']
+    kafka_package_name            = kafkas[0]['package_name']
     cassandra_package_name        = cassandra['package_name']
     cassandra_service_name        = cassandra['service']['name']
     cassandra_num_nodes           = cassandra['nodes']['count']
-    num_producers_per_dispatcher  = int(args['--num-producers-per-dispatcher'])
+    num_producers_per_kafka       = int(args['--num-producers-per-kafka'])
     num_consumers_per_producer    = int(args['--num-consumers-per-producer'])
+    producer_must_fail            = args['--producer-must-fail']
     producer_number_of_words      = int(args['--producer-number-of-words'])
     producer_words_per_second     = int(args['--producer-words-per-second'])
     producer_spark_cores_max      = int(args['--producer-spark-cores-max'])
     producer_spark_executor_cores = int(args['--producer-spark-executor-cores'])
+    consumer_must_fail            = args['--consumer-must-fail']
     consumer_write_to_cassandra   = args['--consumer-write-to-cassandra']
     consumer_batch_size_seconds   = int(args['--consumer-batch-size-seconds'])
     consumer_spark_cores_max      = int(args['--consumer-spark-cores-max'])
     consumer_spark_executor_cores = int(args['--consumer-spark-executor-cores'])
 
-    producer_must_fail = args['--producer-must-fail']
-    consumer_must_fail = args['--consumer-must-fail']
-
-    log.info("Dispatchers: \n{}".format("\n".join(dispatchers)))
+    num_kafkas = len(kafkas)
+    num_dispatchers = len(dispatchers)
+    num_producers = num_kafkas * num_producers_per_kafka
+    num_consumers = num_producers * num_consumers_per_producer
+    num_jobs = num_producers + num_consumers
+    avg_num_jobs_per_dispatcher = num_jobs / num_dispatchers
+    max_num_jobs_per_dispatcher = math.ceil(avg_num_jobs_per_dispatcher)
```

Comment:
Since we are doing integer division here, `math.ceil` will not have the desired result. (or has python changed its behaviour). Looks like it.

Reply:
I'm not sure what you mean.

Reply:
In Python 2:
```python
Python 2.7.15 (default, May  1 2018, 16:44:08)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import math
>>> a = 5 / 3
>>> print a, math.ceil(a)
1 1.0
>>>
```
(`a` is calculated using integer division)

In Python 3:
```python
Python 3.6.4 (default, Mar  9 2018, 23:15:03)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import math
>>> a = 5 / 3
>>> print(a, math.ceil(a))
1.6666666666666667 2
>>>
```